### PR TITLE
Fix ChoiceLoader issue SF2.7+

### PR DESCRIPTION
**Pull request by Mangetsu**
Add ModelChoiceLoader class support for sf2.7+ for multi-choice select widget in Sonata Admin Form

### CODE REVIEW

**Comment by Mangetsu** [comment]
@greg0ire : always issues, now with phpdoc :confused: I'm sure now Sonata don't like me at all

**Comment by greg0ire** [comment]
```diff
index 7f50ef4..9e5c7bf 100644
--- a/Form/DataTransformer/ModelsToArrayTransformer.php
+++ b/Form/DataTransformer/ModelsToArrayTransformer.php
@@ -46,8 +46,9 @@ class ModelsToArrayTransformer implements DataTransformerInterface
      * ModelsToArrayTransformer constructor.
      *
      * @param ModelChoiceList|LazyChoiceList|ModelChoiceLoader $choiceList
-     * @param ModelManagerInterface $modelManager
+     * @param ModelManagerInterface                            $modelManager
      * @param $class
+     *
      * @throws RuntimeException
      */
```

**Comment by Mangetsu** [comment]
@greg0ire : you are talking about this whitespace ? 
`* @param ModelManagerInterface                            $modelManager`

because i dont have it my end that's the code : 

capture2

**Comment by greg0ire** [comment]
Yes, and about the missing white line under `$class`

**Comment by greg0ire** [comment]
That's the problem, you should have it.

**Comment by Mangetsu** [comment]
well I guess i've been something else than a Symfony developer all those past years ^^

thank you very much for the assitance @greg0ire 
btw i managed to get php cs fixer with phpstorm over windows like in the attached image if anyone faces the same issue : 

capture

**Comment by greg0ire** [comment]
Yay! Green build! One last (I promise), very small thing : let's get rid of the merge commit.

```
git rebase origin/master
git push --force
```

**Comment by Mangetsu** [comment]
It's done sir :D hope I keep the green :dancers:

**Comment by greg0ire** [comment]
Argh sorry, wrong promise, what do you get when you run `git remote -vvv` ?

**Comment by Mangetsu** [comment]
what about the SimpleChoiceList called in ModelChoiceList ? the 
the file :  Symfony\Component\Form\Extension\Core\ChoiceList\SimpleChoiceList is removed but still called ModelChoiceList, maybe it should be removed ?

**Comment by greg0ire** [comment]
> what about the SimpleChoiceList called in ModelChoiceList ? the
the file : Symfony\Component\Form\Extension\Core\ChoiceList\SimpleChoiceList is removed but still called ModelChoiceList, maybe it should be removed ?

I don't know about that. Can you answer the question ?

> Argh sorry, wrong promise, what do you get when you run git remote -vvv ?

**Comment by Mangetsu** [comment]
I get this :
origin https://github.com/Mangetsu/SonataAdminBundle.git (fetch)
origin https://github.com/Mangetsu/SonataAdminBundle.git (push)
sonata-project https://github.com/sonata-project/SonataAdminBundle.git (fetch)
sonata-project https://github.com/sonata-project/SonataAdminBundle.git (push)

**Comment by greg0ire** [comment]
Okay, so what I actually meant was : 

```
git rebase sonata-project/master
git push --force
```

**Comment by greg0ire** [comment]
That's better :)

**Comment by greg0ire** [comment]
> what about the SimpleChoiceList called in ModelChoiceList ? the
the file : Symfony\Component\Form\Extension\Core\ChoiceList\SimpleChoiceList is removed but still called ModelChoiceList, maybe it should be removed ?

So, what did you mean by that exactly?

**Comment by Mangetsu** [comment]
@greg0ire : never mind, it can't be touched to avoid regressions with old symfony release (I'm using SF3 and this file does no longer exist : Symfony\Component\Form\Extension\Core\ChoiceList\SimpleChoiceList

it's called and used in the SonataAdminBundle\Form\ChoiceList\ModelChoiceList.php

**Comment by greg0ire** [comment]
Bonus round : if you know how to use `git rebase --interactive`, it would be great that you fixup the last commit into the first one, so

- `git rebase --interactive sonata-project/master`
- move the last line under the first line
- change `pick` to `fixup` (or simply `f`)
- close editor
- `git push --force`

**Comment by greg0ire** [comment]
@OskarStark @core23 this is @Mangetsu first contribution, can you look at it?

**Comment by Mangetsu** [comment]
@greg0ire : something wrong with the pull request ?

**Comment by greg0ire** [comment]
Well you can always improve by applying [this comment](https://github.com/sonata-project/SonataAdminBundle/pull/3655#issuecomment-196898221), but the PR is good enough IMO. Just leave mergers some time to review it.

**Comment by core23** [comment]
LGTM

@OskarStark Can you review this too?

File: Form/Type/ModelType.php [hunk 1]
```diff
@@ -51,7 +51,10 @@ public function buildForm(FormBuilderInterface $builder, array $options)
     {
         if ($options['multiple']) {
             if (array_key_exists('choice_loader', $options) && $options['choice_loader'] !== null) { // SF2.7+
-                $builder->addViewTransformer(new ModelsToArrayTransformer($options['choice_list'], $options['model_manager'], $options['class']), true);
```
Comment (OskarStark) [inline Form/Type/ModelType.php]: this is only a formatting change, isn't it? it confused me a little bit, can we do this in a separate PR?

Comment (OskarStark) [inline Form/Type/ModelType.php]: sorry my fault   ...

**Comment by OskarStark** [comment]
thank you @Mangetsu

**Comment by greg0ire** [comment]
Congrats for your first contribution @Mangetsu ! Keep on the good work!
